dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/net/info/CallbackNetworkInfoProvider.kt

```diff
@@ -60,19 +61,31 @@ internal class CallbackNetworkInfoProvider :
             }
         } catch (e: SecurityException) {
             // RUMM-852 On some devices we get a SecurityException with message
-            // "package does not belong to 10411"
+            // "package does not belong to xxxx"
+            devLogger.e(ERROR_REGISTER, e)
+            networkInfo = NetworkInfo(NetworkInfo.Connectivity.NETWORK_OTHER)
+        } catch (e: RuntimeException) {
+            // RUMM-918 in some cases the device throws a IllegalArgumentException on register
+            // "Too many NetworkRequests filed" This happens when registerDefaultNetworkCallback is
+            // called too many times without matching unregisterNetworkCallback
             devLogger.e(ERROR_REGISTER, e)
             networkInfo = NetworkInfo(NetworkInfo.Connectivity.NETWORK_OTHER)
         }
     }
 
+    @Suppress("TooGenericExceptionCaught")
     override fun unregister(context: Context) {
         val connMgr = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
         try {
             connMgr.unregisterNetworkCallback(this)
         } catch (e: SecurityException) {
             // RUMM-852 On some devices we get a SecurityException with message
-            // "package does not belong to 10411"
+            // "package does not belong to xxxx"
+            devLogger.e(ERROR_UNREGISTER, e)
+        } catch (e: RuntimeException) {
+            // RUMM-918 in some cases the device throws a IllegalArgumentException on register
```

Comment:
Why do you need it also for `unregister` ? Also that comment there is not relevant in this case.

Reply:
Unregister can fail in some rare cases (when you unregister a callback not already registered. I'll fix the comments